subworkflows/local/prepare_references.nf

```diff
@@ -2,115 +2,163 @@
+                }
+                .set{ch_gtf_mix}
+
+        ch_gtf_final = ch_gtf_mix.uncompressed.mix(GUNZIP_GTF.out.gunzip.collect())
+
+        // If no star index, create it
+        BUILD_STAR_GENOME ( ch_fasta_final, ch_gtf_final )
+        // Untar star index if necessary
+        UNTAR_STAR_INDEX(ch_star_index_input)
+        ch_star_index_input.
+            branch{ it ->
+                compressed: it[1].toUriString().endsWith(".gz") // If the file ends with .gz
+                    return [it[0], it[1]]
+                uncompressed: !(it[1].toUriString().endsWith(".gz")) // If the file doesn't end with .gz
+                    return [it[0], it[1]]
+                }
+                .set{ch_star_mix}
+
+        ch_star_mixed = ch_star_mix.uncompressed.mix(UNTAR_STAR_INDEX.out.untar.collect())
+        ch_star_final = ch_star_mixed.mix(BUILD_STAR_GENOME.out.index.collect())
 
-        ch_star = star_index ?
-            Channel.fromPath(star_index).collect().map { it -> [ [id:it[0].simpleName], it ] }
-            : Channel.empty()
-
-        BUILD_STAR_GENOME (ch_fasta, ch_gtf )
-        UNTAR_STAR_INDEX( ch_star.map { it -> [[:], it[1]] } )
-        ch_star_index = (!star_index) ?  BUILD_STAR_GENOME.out.index.collect() :
-                                        (star_index.endsWith(".gz") ? UNTAR_STAR_INDEX.out.untar.map { it[1] } : ch_star)
         // Convert gtf to refflat for picard
-        GTF_TO_REFFLAT(ch_gtf)
+        GTF_TO_REFFLAT(ch_gtf_final)
 
         // Get rRNA transcripts and convert to interval_list format
-        GET_RRNA_TRANSCRIPTS(ch_gtf)
-
+        GET_RRNA_TRANSCRIPTS(ch_gtf_final)
         BEDTOINTERVALLIST( GET_RRNA_TRANSCRIPTS.out.bed.map { it -> [ [id:it.name], it ] }, ch_dict )
         ch_interval = BEDTOINTERVALLIST.out.interval_list.map{ meta, interv -> [interv] }.collect()
 
-        UNTAR_VEP_CACHE (ch_vep_cache)
-        ch_untar_vep = UNTAR_VEP_CACHE.out.untar.map{ meta, files -> [files] }.collect()
-
         // Preparing transcript fasta
-        ch_fasta_fai = ch_fasta.mix(ch_fai.map{meta, fai -> fai}).collect()
-
-        GFFREAD(ch_gtf, ch_fasta_fai)
-        transcript_fasta_no_meta = (!transcript_fasta) ? GFFREAD.out.tr_fasta.collect() :
-                                    (transcript_fasta.endsWith(".gz") ? GUNZIP_TRFASTA.out.gunzip.collect().map{ meta, fasta -> [ fasta ] } : transcript_fasta)
-
-        // Setting up Salmon index
-        ch_salmon = salmon_index ? Channel.fromPath(salmon_index).collect() : Channel.empty()
-        UNTAR_SALMON_INDEX( ch_salmon.map { it -> [[:], it] } )
-        SALMON_INDEX(ch_fasta_no_meta, transcript_fasta_no_meta)
-
-        ch_salmon_index = (!salmon_index) ? SALMON_INDEX.out.index.collect() :
-                                            (salmon_index.endsWith(".gz") ? UNTAR_SALMON_INDEX.out.untar.map { it[1] } : salmon_index.collect())
+        ch_fasta_fai = ch_fasta_final.mix(ch_fai.map{meta, fai -> fai}).collect()
```

Comment:
I would suggest doing a `join` here instead of a mix. The output of mix could be in any order so you might sometimes get [meta, fai, fasta] or something else.